src/jetson/db/file_transfer.py

```diff
@@ -1,47 +1,58 @@
-            self.transport.connect(username=config.FTPUSER, password=config.FTPPASS)
-            self.sftp = paramiko.SFTPClient.from_transport(self.transport)
-            print('Sucessfully connected to host machine')
-
-        except Exception as e:
-            print(e)
-            print('Error connecting to the host machine')
-
-    def transfer(self, input_dir, output_dir):
-        start_time = time.time()                                                                    #time_measurement
-        #make sure directory paths are in correct format
-        if not input_dir.endswith('/'):
-            input_dir = input_dir + '/'
-        if not output_dir.endswith('/'):
-            output_dir = output_dir + '/'
-
-        #transfer files from input directory to output directory
-        try:
-            for image in os.listdir(input_dir):
-                #only transfer files that end with certain type(.py for testing purposes)
-                if image.endswith('.py'):
-                    self.sftp.put(input_dir + image, output_dir + image)
-                    #os.remove(input_dir + image) #remove the image from client side after successful transfer
-            print("The transfer took %s seconds." % (time.time() - start_time))                     #time_measurement
-            
-        
-        except Exception as e:
-            print(e)
-    
+
+@contextmanager
+def ftp_transfer():
+    """Sets up connection to target machine using SFTP and returns transfer function
+
+    Example Usage:
+        with ftp_transfer() as transfer:
+            transfer("./input_dir", "./output_dir", "file_name.ext")
+
+    Yields:
+        [function]: transfers file to output dir using SFTP
+
+        Args:
+            input_dir [string]: input directory for image, can be absolute or relative path
+            output_dir [string]: output directory for image on target machine, can be absolute or relative path
+            image_name [string]: name of file, include file type extension
+    """
+    transport = None
+    sftp = None
+    config = get_config()
     
-    def disconnect(self):
-        
-        #disconnect sftp session
-        self.sftp.close()
-        self.transport.close()
+    try:
+        transport = paramiko.Transport((config["FTPHOST"], 22))
+        transport.connect(username=config["FTPUSER"], password=config["FTPPASS"])
+        sftp = paramiko.SFTPClient.from_transport(transport)
+        print('Sucessfully connected to host machine')
+
+        def transfer(input_dir, output_dir, image_name):
+            # ensure correct format for directory paths
+            if not output_dir.endswith('/'):
+                output_dir = output_dir + '/'
+
+            # transfer files from input directory to output directory
+            try:
+                # only transfer files that end with certain type(.jpg for testing purposes)
+                if input_dir.endswith('.py'):
```

Comment:
this is most likely a bug.  I dont think a directory should end in `.py` please fix.